packages/runtime/lifecycle/src/api.ts

```diff
@@ -32,33 +45,76 @@ export type ApplyInputsRequest = {
   invalidHereafter?: ISO8601;
 };
 /**
- * TODO: comment
+ * This Interface provides capabilities for runnning a Contract over Cardano.
  */
 export interface ContractsAPI {
   /**
-   * TODO: comment
+   * Submit to the Cardano Ledger, the Transaction(Tx) that will create the Marlowe Contract passed in the request.
+   * @param request Request parameters for creating a Marlowe Contract on Cardano
    * @throws DecodingError
    */
-  create(req: CreateRequest): Promise<ContractId>;
-  // [[apply-inputs-next-provider]]
-  // DISCUSSION: Instead of providing a function from "Next" => ApplyInputsRequest, I think
-  //             this as a low level API should just receive the ApplyInputsRequest, and if we want we can
-  //             add a "higher-level" helper
-  //             Search for [apply-inputs-next-provider] for usage
+  submitCreateTx(request: CreateRequest): Promise<[ContractId, TxId]>;
+
   /**
-   * TODO: comment
+   * @experimental
+   * Submit to the Cardano Ledger & Wait Confirmation, the Transaction(Tx) that will create the Marlowe Contract passed in the request.
+   * It's higher level capabilities compared to `submitCreateTx`.
+   * It combines `submitCreateTx` & waitConfirmation for you to abstract Cardano infrastructure concerns.
+   * @param request Request parameters for creating a Marlowe Contract on Cardano
+   * @throws DecodingError
+   */
+  create(request: CreateRequest): Promise<ContractId>;
```

Comment:
The naming is a little confusing, I wrote a different comment but then re-read the doc and code and notice something weird.

From the names I would expect this to make the create transaction and submitCreateTx to do the submission. From the docs, I would expect `submitCreateTx` to just be `create` and this to be `createAndWait`. Moreover, we don't need that, it is a single call away to wait to the Tx.

Reply:
ok, I'm ok to remove it...